File: admin-token-activation-generator/generator.py
import hashlib
import json
import base64
import warnings

warnings.filterwarnings("ignore", category=DeprecationWarning)
import os
import sys

class Colors:
    RESET = "\033[0m"
    BRIGHT = "\033[1m"
    GREEN = "\033[32m"
    YELLOW = "\033[33m"
    CYAN = "\033[36m"
    MAGENTA = "\033[35m"
    RED = "\033[31m"
    WHITE = "\033[37m"

SECRET_SALT = 'magxxicVox_Super_Secure_Salt_2024'

def encode_obf(data):
    json_data = json.dumps(data, separators=(',', ':'))
    b64_data = base64.b64encode(json_data.encode()).decode()
    return b64_data[::-1]

def ask(query):
    return input(f"{Colors.CYAN}│ {Colors.RESET}{query}")

def clear():
    os.system('cls' if os.name == 'nt' else 'clear')

def main():
    while True:
        clear()
        print(f"{Colors.MAGENTA}┌───────────────────────────────────────────────────┐{Colors.RESET}")
        print(f"{Colors.MAGENTA}│{Colors.BRIGHT}{Colors.CYAN}      magxxicVox Admin Toolkit (Python)            {Colors.MAGENTA}│{Colors.RESET}")
        print(f"{Colors.MAGENTA}├───────────────────────────────────────────────────┤{Colors.RESET}")
        print(f"{Colors.MAGENTA}│{Colors.WHITE}  1. Generate Activation Token (from HWID)         {Colors.MAGENTA}│{Colors.RESET}")
        print(f"{Colors.MAGENTA}│{Colors.WHITE}  2. Create AWS Config (aws.sys)                   {Colors.MAGENTA}│{Colors.RESET}")
        print(f"{Colors.MAGENTA}│{Colors.WHITE}  3. Create Brevo Config (brevo.sys)               {Colors.MAGENTA}│{Colors.RESET}")
        print(f"{Colors.MAGENTA}│{Colors.WHITE}  4. Create Mailgun Config (mailgun.sys)           {Colors.MAGENTA}│{Colors.RESET}")
        print(f"{Colors.MAGENTA}│{Colors.WHITE}  5. Create SendGrid Config (sendgrid.sys)         {Colors.MAGENTA}│{Colors.RESET}")
        print(f"{Colors.MAGENTA}│{Colors.WHITE}  6. Create DKIM Config (dkim.sys)                 {Colors.MAGENTA}│{Colors.RESET}")
        print(f"{Colors.MAGENTA}│{Colors.WHITE}  7. Create Direct MX Config (direct_mx.sys)       {Colors.MAGENTA}│{Colors.RESET}")
        print(f"{Colors.MAGENTA}│{Colors.WHITE}  0. Exit                                          {Colors.MAGENTA}│{Colors.RESET}")
        print(f"{Colors.MAGENTA}└───────────────────────────────────────────────────┘{Colors.RESET}")

        choice = ask('Select option: ')

        if choice == '1':
            hwid = ask('Enter User HWID: ')
            token = hashlib.sha256((hwid.strip() + SECRET_SALT).encode()).hexdigest().upper()
            print(f"\n{Colors.GREEN}[+] Generated Token:{Colors.RESET} {Colors.BRIGHT}{token}{Colors.RESET}\n")

        elif choice == '2':
            aws = {
                'region': ask('Region (e.g. us-east-1): '),
                'accessKeyId': ask('Access Key ID: '),
                'secretAccessKey': ask('Secret Access Key: ')
            }
            with open('aws.sys', 'w') as f:
                f.write(encode_obf(aws))
            print(f"\n{Colors.GREEN}[+] Created aws.sys{Colors.RESET}\n")

        elif choice == '3':
            brevo = {
                'user': ask('Brevo User Email: '),
                'apiKey': ask('Brevo API Key: ')
            }
            with open('brevo.sys', 'w') as f:
                f.write(encode_obf(brevo))
            print(f"\n{Colors.GREEN}[+] Created brevo.sys{Colors.RESET}\n")

        elif choice == '4':
            mailgun = {
                'apiKey': ask('Mailgun API Key: '),
                'domain': ask('Mailgun Domain: ')
            }
            with open('mailgun.sys', 'w') as f:
                f.write(encode_obf(mailgun))
            print(f"\n{Colors.GREEN}[+] Created mailgun.sys{Colors.RESET}\n")

        elif choice == '5':
            sendgrid = {
                'apiKey': ask('SendGrid API Key: ')
            }
            with open('sendgrid.sys', 'w') as f:
                f.write(encode_obf(sendgrid))
            print(f"\n{Colors.GREEN}[+] Created sendgrid.sys{Colors.RESET}\n")

        elif choice == '6':
            dkim = {
                'domainName': ask('DKIM Domain: '),
                'keySelector': ask('DKIM Selector: ')
            }
            with open('dkim.sys', 'w') as f:
                f.write(encode_obf(dkim))
            print(f"\n{Colors.GREEN}[+] Created dkim.sys{Colors.RESET}")
            print(f"{Colors.YELLOW}[!] Remember to place dkim_key.pem in the same folder.{Colors.RESET}\n")

        elif choice == '7':
            direct_mx = {
                'retries': int(ask('Max Retries (default 3): ') or 3),
                'timeout': int(ask('Timeout in ms (default 10000): ') or 10000),
                'verifyDns': ask('Verify DNS/MX before send? (y/n): ').lower() == 'y'
            }
            with open('direct_mx.sys', 'w') as f:
                f.write(encode_obf(direct_mx))
            print(f"\n{Colors.GREEN}[+] Created direct_mx.sys{Colors.RESET}\n")

        elif choice == '0':
            sys.exit(0)

        else:
            print(f"{Colors.RED}[!] Invalid selection.{Colors.RESET}")

        ask('Press Enter to continue...')

if __name__ == "__main__":
    main()

File: admin-token-activation-generator/generator.js
'use strict';

process.noDeprecation = true;

const crypto = require('crypto');
const fs = require('fs');
const readline = require('readline');

const colors = {
    reset: "\x1b[0m",
    bright: "\x1b[1m",
    green: "\x1b[32m",
    yellow: "\x1b[33m",
    cyan: "\x1b[36m",
    magenta: "\x1b[35m",
    red: "\x1b[31m",
    white: "\x1b[37m"
};

const SECRET_SALT = 'magxxicVox_Super_Secure_Salt_2024';

const obf = {
    encode: (data) => {
        const json = JSON.stringify(data, null, 4);
        return Buffer.from(json).toString('base64').split('').reverse().join('');
    }
};

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

function ask(query) {
    return new Promise(resolve => rl.question(`${colors.cyan}│ ${colors.reset}${query}`, resolve));
}

async function menu() {
    console.clear();
    console.log(`${colors.magenta}┌───────────────────────────────────────────────────┐${colors.reset}`);
    console.log(`${colors.magenta}│${colors.bright}${colors.cyan}      magxxicVox Admin Toolkit (Node.js)           ${colors.magenta}│${colors.reset}`);
    console.log(`${colors.magenta}├───────────────────────────────────────────────────┤${colors.reset}`);
    console.log(`${colors.magenta}│${colors.white}  1. Generate Activation Token (from HWID)         ${colors.magenta}│${colors.reset}`);
    console.log(`${colors.magenta}│${colors.white}  2. Create AWS Config (aws.sys)                   ${colors.magenta}│${colors.reset}`);
    console.log(`${colors.magenta}│${colors.white}  3. Create Brevo Config (brevo.sys)               ${colors.magenta}│${colors.reset}`);
    console.log(`${colors.magenta}│${colors.white}  4. Create Mailgun Config (mailgun.sys)           ${colors.magenta}│${colors.reset}`);
    console.log(`${colors.magenta}│${colors.white}  5. Create SendGrid Config (sendgrid.sys)         ${colors.magenta}│${colors.reset}`);
    console.log(`${colors.magenta}│${colors.white}  6. Create DKIM Config (dkim.sys)                 ${colors.magenta}│${colors.reset}`);
    console.log(`${colors.magenta}│${colors.white}  7. Create Direct MX Config (direct_mx.sys)       ${colors.magenta}│${colors.reset}`);
    console.log(`${colors.magenta}│${colors.white}  8. Create App Config (config.sys)                ${colors.magenta}│${colors.reset}`);
    console.log(`${colors.magenta}│${colors.white}  0. Exit                                          ${colors.magenta}│${colors.reset}`);
    console.log(`${colors.magenta}└───────────────────────────────────────────────────┘${colors.reset}`);

    const choice = await ask('Select option: ');

    switch (choice) {
        case '1':
            const hwid = await ask('Enter User HWID: ');
            const token = crypto.createHash('sha256').update(hwid.trim() + SECRET_SALT).digest('hex').toUpperCase();
            console.log(`\n${colors.green}[+] Generated Token:${colors.reset} ${colors.bright}${token}${colors.reset}\n`);
            break;

        case '2':
            const aws = {
                region: await ask('Region (e.g. us-east-1): '),
                accessKeyId: await ask('Access Key ID: '),
                secretAccessKey: await ask('Secret Access Key: ')
            };
            fs.writeFileSync('aws.sys', obf.encode(aws));
            console.log(`\n${colors.green}[+] Created aws.sys${colors.reset}\n`);
            break;

        case '3':
            const brevo = {
                user: await ask('Brevo User Email: '),
                apiKey: await ask('Brevo API Key: ')
            };
            fs.writeFileSync('brevo.sys', obf.encode(brevo));
            console.log(`\n${colors.green}[+] Created brevo.sys${colors.reset}\n`);
            break;

        case '4':
            const mailgun = {
                apiKey: await ask('Mailgun API Key: '),
                domain: await ask('Mailgun Domain: ')
            };
            fs.writeFileSync('mailgun.sys', obf.encode(mailgun));
            console.log(`\n${colors.green}[+] Created mailgun.sys${colors.reset}\n`);
            break;

        case '5':
            const sendgrid = {
                apiKey: await ask('SendGrid API Key: ')
            };
            fs.writeFileSync('sendgrid.sys', obf.encode(sendgrid));
            console.log(`\n${colors.green}[+] Created sendgrid.sys${colors.reset}\n`);
            break;

        case '6':
            const dkim = {
                domainName: await ask('DKIM Domain: '),
                keySelector: await ask('DKIM Selector: ')
            };
            fs.writeFileSync('dkim.sys', obf.encode(dkim));
            console.log(`\n${colors.green}[+] Created dkim.sys${colors.reset}`);
            console.log(`${colors.yellow}[!] Remember to place dkim_key.pem in the same folder.${colors.reset}\n`);
            break;

        case '7':
            const dm_config = {
                heloDomain: await ask('HELO/EHLO Domain (e.g. mail.example.com): '),
                timeout: parseInt(await ask('Timeout in ms (default 10000): ')) || 10000
            };
            const dm_settings = {
                retries: parseInt(await ask('Max Retries (default 3): ')) || 3,
                verifyDns: (await ask('Verify DNS/MX before send? (y/n): ')).toLowerCase() === 'y'
            };
            fs.writeFileSync('direct_mx_config.sys', obf.encode(dm_config));
            fs.writeFileSync('direct_mx_settings.sys', obf.encode(dm_settings));
            console.log(`\n${colors.green}[+] Created direct_mx_config.sys and direct_mx_settings.sys${colors.reset}\n`);
            break;

        case '8':
            const sendAttachment = (await ask('Send Attachment? (y/n): ')).toLowerCase() === 'y';
            let attachmentType = 'None';
            if (sendAttachment) {
                console.log('\n  1. PDF');
                console.log('  2. Image');
                console.log('  3. SVG');
                const typeChoice = await ask('Select Attachment Format (1-3): ');
                attachmentType = typeChoice === '1' ? 'PDF' : (typeChoice === '2' ? 'Image' : 'SVG');
            }

            const config = {
                rotateLetters: (await ask('Rotate Letters? (y/n): ')).toLowerCase() === 'y',
                autoShortenLinks: (await ask('Auto Shorten Links? (y/n): ')).toLowerCase() === 'y',
                sendAttachment: sendAttachment,
                attachmentType: attachmentType,
                delayBetweenEmails: parseInt(await ask('Delay between emails (ms): ')) || 2000,
                pauseEvery: parseInt(await ask('Pause every X emails: ')) || 50,
                pauseTime: parseInt(await ask('Pause duration (ms): ')) || 30000,
                encryptionMethod: await ask('Encryption Method (None/AES-256-CBC/ZIP): ') || 'ZIP',
                encryptionPassword: await ask('Encryption Password: ') || 'secret',
                signAttachment: (await ask('Sign Attachment? (y/n): ')).toLowerCase() === 'y'
            };
            fs.writeFileSync('config.sys', obf.encode(config));
            console.log(`\n${colors.green}[+] Created config.sys${colors.reset}\n`);
            break;

        case '0':
            process.exit(0);

        default:
            console.log(`${colors.red}[!] Invalid selection.${colors.reset}`);
    }

    await ask('Press Enter to continue...');
    menu();
}

menu();
